deltachat-ios/Helper/ImageFormat.swift

```diff
@@ -84,25 +84,55 @@ extension ImageFormat {
         return saveImage(data: data, name: name, suffix: suffix)
     }
 
-    public static func saveImage(data: Data, name: String? = nil, suffix: String) -> String? {
+    public static func saveImage(data: Data, name: String? = nil, suffix: String, directory: FileManager.SearchPathDirectory = .applicationSupportDirectory) -> String? {
         var path: URL?
-        guard let directory = try? FileManager.default.url(for: .documentDirectory, in: .userDomainMask,
-                                                              appropriateFor: nil, create: false) as NSURL
-        else { return nil }
+
+        // ensure directory exists (application support dir doesn't exist per default)
+        let fileManager = FileManager.default
+        let urls = fileManager.urls(for: directory, in: .userDomainMask) as [URL]
+        guard let identifier = Bundle.main.bundleIdentifier else {
+            print("err: Could not find bundle identifier")
+            return nil
+        }
+        guard let directoryURL = urls.last else {
```

Comment:
why the last here? https://programmingwithswift.com/save-file-to-documents-directory-with-swift/ uses the first ... just asking, i am not sure which one is better :)

Reply:
I'll use the first one, it is also mentioned in the apple example https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/AccessingFilesandDirectories/AccessingFilesandDirectories.html